### DISCUSSION TITLE
Is `/home/piku/.profile` not used during the deployment?

### DISCUSSION THREAD

**Discussion by osbre** [piku/piku#335] [Q&A]
Hello. I installed [Asdf](https://asdf-vm.com/guide/getting-started.html) for the `piku` user, as that seems to be the best to get the latest Ruby working in Ubuntu. 

It works if I explicitly log in as `piku`:
```shell
$ ssh root@myserver
# su - piku
$ ruby -v
ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [aarch64-linux]
$ gem -v
3.5.3
$ bundle -v
Bundler version 2.5.3
$
```

However, it is not being detected during the deployment:

At first, it says 

```
-----> Checking requirements: ['ruby', 'gem', 'bundle']
[None, None, None]
-----> Generic app detected.
```

In the end, it does:

```
-----> Releasing
/usr/bin/env: 'ruby': No such file or directory
/usr/bin/env: 'ruby': No such file or directory
/usr/bin/env: 'ruby': No such file or directory
-----> Exiting due to release command error value: 127
```

Since the `asdf` has already been added to the `/home/piku/.profile` file and it doesn't work, I have no clue how to get it working. 

Here's what I added to `.profile`, for instance:
```
export ASDF_DIR="$HOME/.asdf"
. "$HOME/.asdf/asdf.sh"
```

It'd be great to find a way how to get `asdf` working, as that would open the door for more tech stacks to be used. (and in this case, to get the Rails stack working)

Thank you

**Comment by chr15m** [comment]
We could add an ENV file option like `SOURCE_PROFILE`, or just make it the default. What do you think @rcarmo?

**Reply by osbre** [reply]
As a user, it seems convenient/intuitive to have it by default, but do what is best for the project.
